packages/web3torrent/notes/channels-integration.md

```diff
@@ -0,0 +1,45 @@
+# How are channels integrated into web3Torrent?
+
+Web3Torrent uses state channels tech to stream money between webTorrent peers, so that downloaders may pay uploaders for data. The granularity of the payments can be tuned, which alters the trust-performance tradeoff. For example, peers can pay per byte (low trust, high overhead for processing payments resulting in low data transfer speeds) or per Megabyte (higher trust, lower overhead and greater transfer speed).
+
+These tradeoffs are set in [`constants.ts`](https://github.com/statechannels/monorepo/blob/master/packages/web3torrent/src/constants.ts).
+
+## Payment channel client
+
+Web3Torrent includes a general purpose, 2-party unidirectional payment channel client that wraps the state channels wallet provider and provides convenient methods such as `sendPayment`. This client takes care of all downstream encoding to interface with the state channels stack.
+
+## Paid streaming extension
+
+Web3Torrent consists of a paid streaming extension to the webTorrent protocol. The purpose of the extension is
+
+- to keep track of peers that have requested data from me, and to maintain a buffer representing the number of bytes I am willing to upload to them
+- to interface with the state channels wallet in order to setup and teardown payment channels with my peers
+- to send and recieve crypto-money in those payment channels in exchange for data
```

Comment:
The current implementation is the extension is simply emitting the events up a layer and let the torrent do all the peer tracking and channel interfacing work. Should we just rename this section `Paid streaming extension` to `Web3Torrent paid streaming client`.

Reply:
91fad3d